pkg/dumper/mysql/dumper.go

```diff
@@ -69,76 +67,46 @@ func (p *myDumper) Close() error {
 	return p.conn.Close()
 }
 
-func (p *myDumper) insertIntoTable(txn *sql.Tx, tableName string, rowChan <-chan database.Row) (int64, error) {
-	columns, err := p.reader.GetColumns(tableName)
-	if err != nil {
+func (p *myDumper) insertIntoTable(txn *sql.Tx, tableName string, rowChan <-chan *database.Table) (int64, error) {
+	var inserted int64
+
+	log.Debug("executing set foreign_key_checks for load data")
+	if _, err := txn.Exec("SET foreign_key_checks = 0;"); err != nil {
+		log.WithError(err).Error("Could not set foreign_key_checks for query")
 		return 0, err
 	}
 
-	// Create query
-	columnsQuoted := make([]string, len(columns))
-	for i, column := range columns {
-		columnsQuoted[i] = p.quoteIdentifier(column)
-	}
-	query := fmt.Sprintf(
-		"LOAD DATA LOCAL INFILE 'Reader::%s' INTO TABLE %s FIELDS TERMINATED BY ',' ENCLOSED BY '\"' ESCAPED BY '\"' (%s)",
-		tableName,
-		p.quoteIdentifier(tableName),
-		strings.Join(columnsQuoted, ","),
-	)
+	for {
+		table, more := <-rowChan
+		if !more {
+			break
+		}
 
-	// Write all rows as csv to the pipe
-	rowReader, rowWriter := io.Pipe()
-	var inserted int64
-	go func(writer *io.PipeWriter) {
-		defer writer.Close()
-
-		w := csv.NewWriter(writer)
-		defer w.Flush()
-
-		for {
-			row, more := <-rowChan
-			if !more {
-				break
-			}
-
-			// Put the data in the correct order and format
-			rowValues := make([]string, len(columns))
-			for i, col := range columns {
-				switch v := row[col].(type) {
-				case nil:
-					rowValues[i] = "NULL"
-				case string:
-					rowValues[i] = row[col].(string)
-				case []uint8:
-					rowValues[i] = string(row[col].([]uint8))
-				default:
-					log.WithField("type", v).Info("we have an unhandled type. attempting to convert to a string \n")
-					rowValues[i] = row[col].(string)
-				}
-			}
-
-			if err := w.Write(rowValues); err != nil {
-				log.WithError(err).Error("error writing record to mysql")
-			}
-
-			atomic.AddInt64(&inserted, 1)
+		insert := sq.Insert(table.Name).SetMap(p.toSQLColumnMap(table.Row))
```

Comment:
And there go's our performance again but that's ok i guess since it would allow proper usage for binary fields without any extra processing.